staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/draplugin.go

```diff
@@ -162,34 +171,74 @@ func NodeV1alpha3(enabled bool) Option {
 	}
 }
 
+// KubeClient grants the plugin access to the API server. This is needed
+// for syncing ResourceSlice objects. It's the responsibility of the DRA driver
+// developer to ensure that this client has permission to read, write,
+// patch and list such objects. It also needs permission to read node objects.
+// Ideally, a validating admission policy should be used to limit write
+// access to ResourceSlices which belong to the node.
+func KubeClient(kubeClient kubernetes.Interface) Option {
+	return func(o *options) error {
+		o.kubeClient = kubeClient
+		return nil
+	}
+}
+
+// NodeName tells the plugin on which node it is running. This is needed for
+// syncing ResourceSlice objects.
+func NodeName(nodeName string) Option {
+	return func(o *options) error {
+		o.nodeName = nodeName
+		return nil
+	}
+}
+
 type options struct {
 	logger                     klog.Logger
 	grpcVerbosity              int
 	driverName                 string
+	nodeName                   string
 	draEndpoint                endpoint
 	draAddress                 string
 	pluginRegistrationEndpoint endpoint
 	unaryInterceptors          []grpc.UnaryServerInterceptor
 	streamInterceptors         []grpc.StreamServerInterceptor
+	kubeClient                 kubernetes.Interface
 
 	nodeV1alpha3 bool
 }
 
 // draPlugin combines the kubelet registration service and the DRA node plugin
 // service.
 type draPlugin struct {
-	registrar *nodeRegistrar
-	plugin    *grpcServer
+	ctx        context.Context // For new background activities that are started later.
+	wg         sync.WaitGroup
+	cancel     func(cause error)
+	registrar  *nodeRegistrar
+	plugin     *grpcServer
+	driverName string
+	nodeName   string
+	kubeClient kubernetes.Interface
+
+	// Information about resource publishing changes concurrently and thus
+	// must be protected by the mutex.
+	mutex                   sync.Mutex
+	publishResources        bool
+	nodeResources           []*resourceapi.ResourceModel
+	resourceSliceController *resourceslice.Controller
 }
 
 // Start sets up two gRPC servers (one for registration, one for the DRA node
 // client). By default, all APIs implemented by the nodeServer get registered.
-func Start(nodeServer interface{}, opts ...Option) (result DRAPlugin, finalErr error) {
-	d := &draPlugin{}
-
+//
+// The context and/or DRAPlugin.Stop can be used to stop all background activity.
+// Stop also blocks. A logger can be stored in the context to add values or
+// a name to all log entries.
+func Start(ctx context.Context, nodeServer interface{}, opts ...Option) (result DRAPlugin, finalErr error) {
+	logger := klog.FromContext(ctx)
 	o := options{
 		logger:        klog.Background(),
-		grpcVerbosity: 4,
+		grpcVerbosity: 6, // Logs requests and responses, which can be large.
```

Comment:
Do we really want this hard coded like this?

Reply:
This is just the default. It can be reconfigured with `GRPCVerbosity`. E2E testing uses this to increase the verbosity without having to add `-v=6` and then run everything with that higher verbosity.

Reply:
I bumped up the default because I found the output to be a bit too much for a cluster running at `-v=5`.

Reply:
Ack